docs/TransparentLogging.md

```diff
@@ -0,0 +1,280 @@
+technically-invalid objects that are still signed and distributed can be
+monitored and attributed.
+
+ - A Certificate Transparency Log may well accept X.509 certificates that fail
+   various sorts of validity check, as long as the signature validation is
+   correct.  This means that certificates that might be accepted by lax clients
+   are still logged, and certificate authorities that issue invalid certificates
+   can be held to account more easily.
+
+## Inclusion Proofs vs. Promises
+
+Transparent Logs have two distinct mechanisms for guaranteeing inclusion of a
+particular entry in the Log.
+
+ - An **inclusion proof** gives cryptographic proof that a particular Merkle
+   tree head includes a given entry: the entry's hash and the proof hashes can be
+   combined to recreate the root hash of the tree.
+ - An **inclusion promise** is a signed commitment from the Log to incorporate a
+   particular entry within a defined time period.  This requires a separate
+   subsequent check to confirm that the Log does indeed fulfil this promise.
+
+A related factor is the number of entries that are incorporated into each new
+tree head issued by the Log, which we'll call the tree head **batch size**.
+
+ - Logs that issue inclusion promises can batch together multiple submissions
+   (e.g. an hour's worth) into a new tree head.  This is more likely to produce
+   large batch sizes, but is not guaranteed to (e.g. if submission rates are
+   very low).
+ - Logs that immediately issue inclusion proofs are likely to have smaller batch
+   sizes, either because they issue a new tree head per submission (batch size =
+   1), or because the amount of batching possible while still responding to
+   submissions quickly is small.
+
+The batch size per new tree head is important for privacy reasons: a small batch
+size, particularly a batch of size 1, means that a new signed tree head
+correlates directly with a single submission.  This in turn means that a user
+requesting proofs to/from that tree head, or gossiping that tree head, is likely
+to be interested in that specific entry in the Log.
+
+So, if privacy is a concern then:
+ - if submission rates are very low, then using inclusion promises with a large
+   batch size is most likely to preserve privacy
+ - if submission rates are very high, and some latency on submissions is
+   acceptable, then just issuing inclusion proofs may be feasible.
+
+A new transparent Log should consider whether it requires both inclusion
+promises and proofs, or just the latter, based on assessing the concerns above
+together with other factors:
+
+ - Inclusion promises allow more operational flexibility; a Log can continue to
+   issue promises even if its Merkle tree storage is temporarily unavailable,
+   and the (potentially) larger batching involved may allow for a more efficient
+   implementation.
+     - However, issuing a promise that cannot subsequently be fulfilled
```

Comment:
I'd be tempted to embolden that **However** :)

Reply:
Tried it but I thought it looked odd.